geoh5py/shared/merging/base.py

```diff
@@ -0,0 +1,200 @@
+#  Copyright (c) 2023 Mira Geoscience Ltd.
+#
+#  This file is part of geoh5py.
+#
+#  geoh5py is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU Lesser General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  geoh5py is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public License
+#  along with geoh5py.  If not, see <https://www.gnu.org/licenses/>.
+
+from abc import ABC, abstractmethod
+from typing import cast
+
+import numpy as np
+
+from ...data import Data
+from ...objects import ObjectBase
+from ...workspace import Workspace
+
+
+class BaseMerger(ABC):
+    _type: type = ObjectBase
+
+    def __init__(self):
+        """
+        Merging class for :obj:geoh5py.objects.ObjectBase of the same types.
+        """
+
+    @classmethod
+    def merge_data(cls, input_entities: list[ObjectBase]) -> dict[str, dict]:
```

Comment:
That method is pretty complicated. Consider pre-allocating your arrays from the start, and just insert if the data.name and entity_type.name match, otherwise it's a new vector.

Reply:
split it in a bunch of function.
Tell me if some functions still need to be splitted.